dbus-cxx/sendmsgtransport.cpp

```diff
@@ -108,10 +266,17 @@ std::shared_ptr<SendmsgTransport> SendmsgTransport::create( int fd, bool initial
 }
 
 ssize_t SendmsgTransport::writeMessage( std::shared_ptr<const DBus::Message> message, uint32_t serial ){
```

Comment:
Would it be better to simply provide a different version of `writeMessage` for Windows only so that we don't have a bunch of `#if`s littering the function?  That would probably be a bit clearer.

Reply:
I will re-factor.

Reply:
Re-factored and left a single #if-else.